pkg/operator/usercaobservation/user_ca_observation_controller.go

```diff
@@ -0,0 +1,94 @@
+package usercaobservation
+
+import (
+	"context"
+	"time"
+
+	configinformers "github.com/openshift/client-go/config/informers/externalversions"
+	configlistersv1 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/management"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/openshift/cluster-openshift-controller-manager-operator/pkg/util"
+)
+
+type UserCAObservationController struct {
+	name                 string
+	operatorConfigClient v1helpers.OperatorClient
+	proxyLister          configlistersv1.ProxyLister
+	resourceSyncer       resourcesynccontroller.ResourceSyncer
+	runFn                func(ctx context.Context, workers int)
+	syncCtxt             factory.SyncContext
+}
+
+func NewUserCAObservationController(operatorConfigClient v1helpers.OperatorClient,
+	configInformers configinformers.SharedInformerFactory,
+	resourceSyncer resourcesynccontroller.ResourceSyncer,
+	eventRecorder events.Recorder) *UserCAObservationController {
+	c := &UserCAObservationController{
+		name:                 "UserCAObservationController",
+		operatorConfigClient: operatorConfigClient,
+		proxyLister:          configInformers.Config().V1().Proxies().Lister(),
+		resourceSyncer:       resourceSyncer,
+	}
+	informers := []factory.Informer{
+		operatorConfigClient.Informer(),
+		configInformers.Config().V1().Proxies().Informer(),
+	}
+	f := factory.New().
+		WithSync(c.Sync).
+		WithSyncContext(c.syncCtxt).
+		WithInformers(informers...).
+		ResyncEvery(10*time.Minute).
+		ToController(c.name, eventRecorder.WithComponentSuffix("user-ca-observation-controller"))
+	c.runFn = f.Run
+	return c
+}
+
+func (c *UserCAObservationController) Run(ctx context.Context, workers int) {
+	c.runFn(ctx, workers)
+}
+
+func (c *UserCAObservationController) Sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	operatorSpec, _, _, err := c.operatorConfigClient.GetOperatorState()
+	if err != nil {
+		return nil
+	}
+
+	if !management.IsOperatorManaged(operatorSpec.ManagementState) {
+		return nil
+	}
+
+	source, err := c.findProxyCASource()
+	if err != nil {
+		return err
+	}
+	destination := resourcesynccontroller.ResourceLocation{
+		Namespace: util.TargetNamespace,
+		Name:      "openshift-user-ca",
+	}
+	err = c.resourceSyncer.SyncConfigMap(destination, source)
```

Comment:
so either user ca is a singleton like proxy ca, or SyncConfigMap is doing a merge of some sort 

come clarification either way would be good

also, some details on why we need to copy ... i.e. the specfics around what breaks down with update-ca-trust if we read the "original" global CA vs. the user CA we've copied to would be good

How does the fact that we copy the proxy's trusted CA reconcile, get us out of jail, play nice, etc. with what I read in https://github.com/openshift/api/blob/master/config/v1/types_proxy.go#L44-L69

Reply:
I updated the comment above to clarify what is going on overall, and why we need to copy the user CA to the openshift-controller-manager namespace. We get out of jail because build pods run `update-ca-trust extract` themselves.

My updated commit message indicates why we need to do this (namely, the issue with `keytool` formatted trust bundles).